packages/iTwinUI-react/src/core/Table/hooks/useScrollToRow.ts

```diff
@@ -0,0 +1,67 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import type { Row } from 'react-table';
+import { TableProps } from '../Table';
+
+type ScrollToRow<T extends Record<string, unknown>> = {
+  scrollToIndex: number | undefined;
+  tableRowRef: (row: Row<T>) => (element: HTMLDivElement) => void;
+};
+
+type ScrollToRowProps<T extends Record<string, unknown>> = Pick<
+  TableProps<T>,
+  'data' | 'enableVirtualization' | 'getRowId' | 'paginatorRenderer'
+> & { scrollToItem?: T };
+
+export function useScrollToRow<T extends Record<string, unknown>>({
+  data,
+  enableVirtualization,
+  scrollToItem,
+  getRowId,
+  paginatorRenderer,
+}: ScrollToRowProps<T>): ScrollToRow<T> {
+  const rowRefs = React.useRef<Record<number, HTMLDivElement>>({});
+
+  const index = React.useMemo((): number => {
+    if (!scrollToItem || !getRowId || paginatorRenderer) {
+      return -1;
+    }
+
+    return data.findIndex(
```

Comment:
Maybe instead of `data` we should use `page` here therefore we wouldn't need to call `getRowId` just `row.id` as I see it has 3 parameters that we are not passing. But we also need to make `scrollIntoItem` to be row id instead of row data.

Reply:
Absolutely! I swear I tested against sorted and filtered data before, but must have changed something and now it's off. Using `page` is the right way; `data` causes havoc on sorted or filtered tables.